db/collection.go

```diff
@@ -148,66 +149,100 @@ func (c *Collection) AddIndex(config IndexConfig) error {
 // ReadTxn creates an explicit readonly transaction. Any operation
 // that tries to mutate an instance of the collection will ErrReadonlyTx.
 // Provides serializable isolation gurantees.
-func (c *Collection) ReadTxn(f func(txn *Txn) error) error {
-	return c.db.readTxn(c, f)
+func (c *Collection) ReadTxn(f func(txn *Txn) error, opts ...TxnOption) error {
+	return c.db.readTxn(c, f, opts...)
 }
 
 // WriteTxn creates an explicit write transaction. Provides
 // serializable isolation gurantees.
-func (c *Collection) WriteTxn(f func(txn *Txn) error) error {
-	return c.db.writeTxn(c, f)
+func (c *Collection) WriteTxn(f func(txn *Txn) error, opts ...TxnOption) error {
+	return c.db.writeTxn(c, f, opts...)
 }
 
 // FindByID finds an instance by its ID.
 // If doesn't exists returns ErrNotFound.
-func (c *Collection) FindByID(id core.InstanceID) (instance []byte, err error) {
+func (c *Collection) FindByID(id core.InstanceID, opts ...TxnOption) (instance []byte, err error) {
 	_ = c.ReadTxn(func(txn *Txn) error {
 		instance, err = txn.FindByID(id)
 		return err
-	})
+	}, opts...)
 	return
 }
 
-// Create creates instances in the collection.
-func (c *Collection) Create(vs ...[]byte) (ids []core.InstanceID, err error) {
+// Create creates an instance in the collection.
+func (c *Collection) Create(v []byte, opts ...TxnOption) (ids []core.InstanceID, err error) {
+	_ = c.WriteTxn(func(txn *Txn) error {
+		ids, err = txn.Create(v)
+		return err
+	}, opts...)
+	return
+}
+
+// CreateMany creates multiple instances in the collection.
+func (c *Collection) CreateMany(vs [][]byte, opts ...TxnOption) (ids []core.InstanceID, err error) {
 	_ = c.WriteTxn(func(txn *Txn) error {
 		ids, err = txn.Create(vs...)
 		return err
-	})
+	}, opts...)
 	return
 }
 
-// Delete deletes instances by its IDs. It doesn't
+// Delete deletes an instance by its ID. It doesn't
 // fail if the ID doesn't exist.
-func (c *Collection) Delete(ids ...core.InstanceID) error {
+func (c *Collection) Delete(id core.InstanceID, opts ...TxnOption) error {
+	return c.WriteTxn(func(txn *Txn) error {
+		return txn.Delete(id)
+	}, opts...)
+}
+
+// Delete deletes multiple instances by ID. It doesn't
+// fail if one of the IDs don't exist.
+func (c *Collection) DeleteMany(ids []core.InstanceID, opts ...TxnOption) error {
 	return c.WriteTxn(func(txn *Txn) error {
 		return txn.Delete(ids...)
-	})
+	}, opts...)
 }
 
-// Save saves changes of instances in the collection.
-func (c *Collection) Save(vs ...[]byte) error {
+// Save saves changes of an instance in the collection.
+func (c *Collection) Save(v []byte, opts ...TxnOption) error {
+	return c.WriteTxn(func(txn *Txn) error {
+		return txn.Save(v)
+	}, opts...)
+}
+
+// SaveMany saves changes of multiple instances in the collection.
+func (c *Collection) SaveMany(vs [][]byte, opts ...TxnOption) error {
```

Comment:
This doesn't seem worth it to me to have these different flavors of create and save for single and many. Can we just make the auth an explicit param and not use options and stick with varadic args?

Reply:
Is there a downside other than just having another method? Mongo does this and it feels pretty intuitive as a user IMO.

Reply:
Ok, I can live with that